source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -51,34 +51,19 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
     span.setTag(TracingConstants::get().TraceStatus, TracingConstants::get().TraceOk);
     authz_response->status = CheckStatus::OK;
     if (response->has_ok_response()) {
-      toAuthzResponseHeader(authz_response, response->ok_response().headers());
-      if (response->ok_response().headers_to_remove_size() > 0) {
-        for (const auto& header : response->ok_response().headers_to_remove()) {
-          authz_response->headers_to_remove.push_back(Http::LowerCaseString(header));
-        }
+      const auto& ok_response = response->ok_response();
+      copyHeaderMutationsIntoResponse(authz_response, ok_response.headers(),
+                                      ok_response.response_headers_to_add(),
+                                      ok_response.headers_to_remove());
+
+      for (const auto& query_parameter : ok_response.query_parameters_to_set()) {
+        authz_response->query_parameters_to_set.emplace_back(
```

Comment:
The auth service is sending back an invalid url parameter. Would it be better to drop it, and increment a metric, instead of url-encoding? I'd argue that the service should've already url-encoded it if that was the intention.

What is the current end-to-end behavior if the service sends back a query param that should have been url-encoded? Where do things fail?

Either way, this will need either a runtime guard or config for the behavior change.

Reply:
For clarity: when you say "drop it," you mean drop the client request, yes?

Reply:
I have not tested invalid query parameter behavior, but I know invalid headers are rejected at the codec.

Invalid query parameters would make the path pseudoheader invalid, so I think it would be similar.

I'll test this tomorrow.